contracts/bonding/libraries/TranscoderPool.sol

```diff
@@ -0,0 +1,372 @@
+pragma solidity ^0.4.17;
+
+import "zeppelin-solidity/contracts/math/SafeMath.sol";
+
+
+/*
+ * @title A pool of registered transcoders and their stakes implemented as a sorted doubly linked list
+ */
+library TranscoderPool {
+    using SafeMath for uint256;
+
+    // Represents a transcoder in the list
+    struct Node {
+        uint256 stake;                   // Stake for the transcoder represented by this node
+        address nextTranscoder;          // Address of next transcoder in the list
+        address prevTranscoder;          // Address of previous transcoder in the list
+    }
+
+    // Represents a sorted doubly linked list
+    struct Data {
```

Comment:
Since we're usually using the library by attaching its functions to a struct via the `using A for B` directive I didn't want to shadow the library name by naming the struct the same thing (which the compiler gives you a warning for when the struct is defined within the library). I decided to just give the struct a generic name like `Data` since it'll be referenced as `SortedDoublyLL.Data` in any contract that is using the `SortedDoublyLL` library. I'm definitely not attached to the naming though and am open to better suggestions!